### PR TITLE
Chat: Add Angular AI And Chatbot Integration Demo

### PR DESCRIPTION
**Pull request by Zedwag**

### CODE REVIEW

File: apps/demos/Demos/Chat/AIAndChatbotIntegration/Angular/app/app.service.ts [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+
+    setTimeout(() => {
+        this.copyButtonIcon = 'copy';
+    }, 2500);
+}
+
+  onRegenerateButtonClick() {
+    this.updateLastMessage();
+    this.regenerate();
+  }
```
Comment (ksercs) [inline apps/demos/Demos/Chat/AIAndChatbotIntegration/Angular/app/app.service.ts]: I think too much stuff is in the service now. Everything related to view (dom / UI) should be in the component. Service is for data/utils/API calls.

File: apps/demos/Demos/Chat/AIAndChatbotIntegration/Angular/app/app.service.ts [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+import remarkParse from 'remark-parse';
+import remarkRehype from 'remark-rehype';
+import rehypeStringify from 'rehype-stringify';
+import { 
+  User,
+  Message, 
+  Alert,
+  MessageEnteredEvent
+} from 'devextreme/ui/chat';
+import DataSource from 'devextreme/data/data_source';
```
Comment (ksercs) [inline apps/demos/Demos/Chat/AIAndChatbotIntegration/Angular/app/app.service.ts]: kindly use new imports from `common`

File: apps/demos/Demos/Chat/AIAndChatbotIntegration/Angular/app/app.service.ts (hunk 1 above)
Comment (marker-dao) [reply apps/demos/Demos/Chat/AIAndChatbotIntegration/Angular/app/app.service.ts]: [995d5d8a2f2d789fb73b106bb05189b52e6568df]

File: apps/demos/Demos/Chat/AIAndChatbotIntegration/Angular/app/app.service.ts (hunk 2 above)
Comment (marker-dao) [reply apps/demos/Demos/Chat/AIAndChatbotIntegration/Angular/app/app.service.ts]: [995d5d8a2f2d789fb73b106bb05189b52e6568df, 780dbd2788b9e40f577d05843a55c7c01e01f6eb]

File: apps/demos/Demos/Chat/AIAndChatbotIntegration/Angular/app/app.service.ts [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+
+  store: any[] = [];
+  messages: any[] = [];
+  alerts: Alert[] = [];
+
+  customStore: CustomStore;
+
+  dataSource: DataSource;
+
+  isDisabled: boolean;
```
Comment (ksercs) [inline apps/demos/Demos/Chat/AIAndChatbotIntegration/Angular/app/app.service.ts]: this prop is directly related to UI only, I believe it should be in the component

Comment (marker-dao) [reply apps/demos/Demos/Chat/AIAndChatbotIntegration/Angular/app/app.service.ts]: [d0e9bca937bd1d048529c35595f8b3cdf3eb5616]

**Review by ksercs — APPROVED** [review]

**Pull request merged**